src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services6.yml

```diff
@@ -37,3 +37,5 @@ services:
     new_factory1: { class: FooBarClass, factory: factory}
     new_factory2: { class: FooBarClass, factory: ['@baz', getClass]}
     new_factory3: { class: FooBarClass, factory: [BazClass, getInstance]}
+    new_factory4: { class: BazClass, factory: [~, getInstance]}
+    new_factory5: { class: BazClass, factory: [getInstance]}
```

Comment:
I don't think we need to support this one, it's rather confusing to have an array with just one element. Using `null` seems more discoverable and easy enough to understand.